hpack/hpack.py

```diff
@@ -264,20 +155,20 @@ def add(self, to_add, sensitive, huffman=False):
         indexbit = INDEX_INCREMENTAL if not sensitive else INDEX_NEVER
 
         # Search for a matching header in the header table.
-        match = self.matching_header(name, value)
+        match = self.header_table.search(name, value)
 
         if match is None:
             # Not in the header table. Encode using the literal syntax,
             # and add it to the header table.
             encoded = self._encode_literal(name, value, indexbit, huffman)
             if not sensitive:
-                self._add_to_header_table(to_add)
+                self.header_table.add(name, value)
             return encoded
 
         # The header is in the table, break out the values. If we matched
         # perfectly, we can use the indexed representation: otherwise we
         # can use the indexed literal.
-        index, perfect = match
+        index, hname, perfect = match
```

Comment:
> It looks like there may not be a flow where we return name but set perfect to False and have sensitive be False as well, or we'd have caught this. With that in mind, can we add one?

Well we're saved by:

    name, value = to_add

So name should always equal hname.